src/api/response/iiif/collection.js

```diff
@@ -0,0 +1,125 @@
+const dcUrl = process.env.DC_URL;
+
+async function transform(response, pager) {
+  if (response.statusCode === 200) {
+    const responseBody = JSON.parse(response.body);
+    const pageInfo = await pager.pageInfo(responseBody.hits.total.value);
+
+    return {
+      statusCode: 200,
+      body: JSON.stringify(await buildCollection(responseBody, pageInfo)),
+    };
+  }
+  return transformError(response);
+}
+
+async function buildCollection(responseBody, pageInfo) {
+  const collectionLabel = pageInfo.options.collectionLabel;
+  const collectionSummary = pageInfo.options.collectionSummary;
+
+  return {
+    "@context": ["http://iiif.io/api/presentation/3/context.json"],
+    id: collectionId(pageInfo),
+    type: "Collection",
+    label: {
+      none: [`${collectionLabel}`],
+    },
+    summary: {
+      none: [`${collectionSummary}`],
+    },
+    homepage: [
+      {
+        id: homepageUrl(pageInfo),
+        type: "Text",
+        format: "text/html",
+        label: {
+          none: [`Results for ${collectionLabel} of ${collectionSummary}`],
+        },
+      },
+    ],
+    items: getItems(responseBody?.hits?.hits, pageInfo),
+  };
+}
+
+function getItems(hits, pageInfo) {
+  const items = hits.map((item) => loadItem(item["_source"]));
+
+  if (pageInfo?.next_url) {
+    items.push({
+      id: pageInfo.next_url,
+      type: "Collection",
+      label: {
+        none: ["Next page"],
+      },
+    });
+  }
+
+  return items;
+}
+
+function collectionId(pageInfo) {
+  let id = pageInfo.query_url;
+  if (pageInfo.current_page > 1) {
+    id = id + `?page=${pageInfo.current_page}`;
+  }
+  return id;
+}
+
+function homepageUrl(pageInfo) {
+  return encodeURI(
+    dcUrl +
+      "/search?q=" +
+      '"' +
+      pageInfo.options.queryString.replace(/\"/g, '\\"') +
+      '"'
+  );
```

Comment:
JavaScript has a great built-in [`URL`](https://nodejs.org/docs/latest-v16.x/api/url.html) class that can automatically handle URL path joining and query encoding (and a lot of other things) that's far more reliable than string concatenation:
  - `new URL(relativePath, base)` resolves the full URL and accounts for the fact that the base may or may not have a trailing slash
  - The `URL` class' `searchParams` property has `get`/`set` methods that act on the query part of the URL directly and handle encoding/decoding for you
  - `searchParams` also handles `"` escaping automatically, so the `.replace(/\"/g, '\\"')` can probably go away (unless I'm wrong about what you're trying to do there).
  - An instance of the `URL` class will automatically be stringified when interpolated or converted to JSON, so you don't even need to call `.toString()` on it yourself.

That would leave:
```
const result = new URL("/search", dcUrl);
result.searchParams.set("q", pageInfo.options.queryString);
return result;
```

Reply:
I used this as the most obvious example, but I think there are a few places in the IIIF code where you have to construct URLs, and I think using the `URL` class provides a bit of a formatting and encoding safety net.